server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala

```diff
@@ -130,6 +132,11 @@ class Serializer {
   private implicit val sessionWrites: Writes[Session] = Json.writes[Session]
 
   private implicit val mailboxIdWrites: Writes[MailboxId] = mailboxId => JsString(mailboxId.serialize)
+  private implicit val mailboxIdReads: Reads[MailboxId] = {
+    case JsString(serializedMailboxId) => JsSuccess(mailboxIdFactory.fromString(serializedMailboxId))
```

Comment:
what happens if fromString is failing?

Reply:
The deserialization would fail I guess? I took that from your team's work on event sourcing though (cf https://github.com/linagora/james-project/blob/master/mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala#L241)

So maybe you can explain it to me?

Reply:
(att a catch block, a JsError and a test for it ;-) )

Reply:
I think that we forgot it could throw as it didn't appear in the method signature.
but you should do  `Try(JsSuccess(mailboxIdFactory.fromString(serializedMailboxId)))` and return a JsError in case of failure